general/templatetags/paginator.py

```diff
@@ -48,14 +48,31 @@ def show_paginator(context, paginator, page, current_page, request, anchor="", n
 
     # although paginator objects are 0-based, we use 1-based paging
     page_numbers = [n for n in range(min_page_num, max_page_num) if n > 0 and n <= paginator.num_pages]
-    
+
     params = urllib.urlencode([(key, value.encode('utf-8')) for (key, value) in request.GET.items() if key.lower() != u"page"])
-    
+
     if params == "":
         url = request.path + u"?page="
     else:
         url = request.path + u"?" + params + u"&page="
 
+    url_last_page = None
+
+    # Page could be a dict or an object
+    if isinstance(page, dict):
+        url_prev_page = url + str(page['previous_page_number'])
+        url_next_page =  url + str(page['next_page_number'])
+        url_first_page = url + '1'
```

Comment:
Why we don't have `url_last_page` in this case?
I tested locally and the first thing I tried (clicking on the last page) used `page=None` which obviously does not work...

Reply:
Ok, solved!